File: app/src/mock/java/com/kaciula/archiman/data/FakeUsersDataSource.java
package com.kaciula.archiman.data;

import com.kaciula.archiman.domain.entity.User;
import io.reactivex.Completable;
import io.reactivex.Observable;
import io.reactivex.functions.Action;
import java.util.ArrayList;
import java.util.List;

public class FakeUsersDataSource implements UsersDataSource {

  @Override
  public Observable<List<User>> getUsers() {
    List<User> users = new ArrayList<>(6);
    users.add(User.create(1, "ka"));
    users.add(User.create(2, "andrei"));
    users.add(User.create(3, "arti"));
    users.add(User.create(4, "marius"));
    users.add(User.create(5, "stefan"));
    users.add(User.create(6, "bogdan"));
    return Observable.just(users);
  }

  @Override
  public Completable createOrUpdateUsers(List<User> users) {
    return Completable.fromAction(new Action() {
      @Override
      public void run() throws Exception {
        // Do nothing here
      }
    });
  }
}

File: app/src/dev/java/com/kaciula/archiman/infrastructure/data/fake/FakeUsersDataSource.kt
package com.kaciula.archiman.infrastructure.data.fake

import com.kaciula.archiman.domain.model.User
import com.kaciula.archiman.infrastructure.data.UserDataStore
import io.reactivex.Completable
import io.reactivex.Single
import java.util.*

class FakeUsersDataSource : UserDataStore {

    override fun getUsers(): Single<List<User>> {
        val users = ArrayList<User>(6)
        users.add(User(1, "ka"))
        users.add(User(2, "andrei"))
        users.add(User(3, "arti"))
        users.add(User(4, "marius"))
        users.add(User(5, "stefan"))
        users.add(User(6, "bogdan"))
        return Single.just<List<User>>(users)
    }

    override fun createOrUpdateUsers(users: List<User>): Completable {
        return Completable.fromAction {
            // Do nothing here
        }
    }
}
